smos-report-cursor/src/Smos/Cursor/Report/Next.hs

```diff
@@ -36,34 +41,146 @@ produceNextActionReportCursor dc = do
       .| C.map (uncurry makeNextActionEntryCursor)
       .| sinkList
 
-type NextActionReportCursor = NonEmptyCursor NextActionEntryCursor
+
+data NextActionReportCursor
+  = NextActionReportCursor
+    { nextActionReportCursorNextActionEntryCursors :: NonEmptyCursor NextActionEntryCursor
+    , nextActionReportCursorSelectedNextActionEntryCursors :: Maybe (NonEmptyCursor NextActionEntryCursor)
+    , nextActionReportCursorFilterBar :: TextCursor
+    , nextActionReportCursorFilterValid :: Bool
```

Comment:
We don't need to put this field here, we could just parse the text during rendering to do that.
We need the `nextActionReportCursorSelectedNextActionEntryCursors` so that we can scroll around in it.

Reply:
Okay fab. Thanks for pointing that out!